examples/advanced_operations/add_ad_customizer.rb

```diff
@@ -0,0 +1,383 @@
+  response = client.service.feed_mapping.mutate_feed_mappings(
+    customer_id: customer_id,
+    operations: [operation],
+  )
+
+  # Displays the results.
+  response.results.each do |result|
+    puts "Created feed mapping with resource name: #{result.resource_name}"
+  end
+end
+
+# Creates two different feed items to enable two different ad customizations.
+def create_feed_items(
+  client,
+  customer_id,
+  ad_customizer_feed_resource_name,
+  ad_customizer_feed_attributes)
+  feed_item_operations = []
+
+  feed_item_operations << create_feed_item_operation(
+    client,
+    "Mars",
+    "$1234.56",
+    # Set the date to the 1st of the current month.
+    DateTime.new(Date.today.year, Date.today.month, 1, 0, 0, 0).strftime("%Y%m%d %H%M%S"),
+    ad_customizer_feed_resource_name,
+    ad_customizer_feed_attributes,
+  )
+
+  feed_item_operations << create_feed_item_operation(
+    client,
+    "Venus",
+    "$6543.21",
+    # Set the date to the 15th of the current month.
+    DateTime.new(Date.today.year, Date.today.month, 15, 0, 0, 0).strftime("%Y%m%d %H%M%S"),
+    ad_customizer_feed_resource_name,
+    ad_customizer_feed_attributes,
+  )
+
+  # Adds the feed items.
+  response = client.service.feed_item.mutate_feed_items(
+    customer_id: customer_id,
+    operations: feed_item_operations,
+  )
+
+  feed_item_resource_names = []
+  response.results.each do |result|
+    puts "Created feed item with resource name #{result.resource_name}"
+    feed_item_resource_names << result.resource_name
+  end
+
+  feed_item_resource_names
+end
+
+# Creates a FeedItemOperation.
+def create_feed_item_operation(
+  client,
+  name,
+  price,
+  date,
+  ad_customizer_feed_resource_name,
+  ad_customizer_feed_attributes)
+  client.operation.create_resource.feed_item do |item|
+    item.feed = ad_customizer_feed_resource_name
+    item.attribute_values << client.resource.feed_item_attribute_value do |v|
+      v.feed_attribute_id = ad_customizer_feed_attributes["Name"]
+      v.string_value = name
+    end
+    item.attribute_values << client.resource.feed_item_attribute_value do |v|
+      v.feed_attribute_id = ad_customizer_feed_attributes["Price"]
+      v.string_value = price
+    end
+    item.attribute_values << client.resource.feed_item_attribute_value do |v|
+      v.feed_attribute_id = ad_customizer_feed_attributes["Date"]
+      v.string_value = date
+    end
+  end
+end
+
+# Restricts the feed items to work only with a specific ad group; this prevents
+# the feed items from being used elsewhere and makes sure they are used only for
+# customizing a specific ad group.
+def create_feed_item_targets(
+  client,
+  customer_id,
+  ad_group_ids,
+  feed_item_resource_names)
+  # Bind each feed item to a specific ad group to make sure it will only be
+  # used to customize ads inside that ad group; using the feed item elsewhere
+  # will result in an error.
+  for i in 0..(feed_item_resource_names.size - 1)
```

Comment:
(feed_item_resource_names.size - 1).times do |i|

Reply:
or (0..(feed_item_resource_names.size - 1)).each do |i|

Reply:
The point is just prefer to use an iterator rather than a normal "loop" like in other languages.

Reply:
Changed to using `times`.